components/org.wso2.carbon.identity.api.user.session/org.wso2.carbon.identity.api.user.session.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/session/v1/impl/UserIdApiServiceImpl.java

```diff
@@ -61,11 +65,27 @@ public Response terminateSessionBySessionId(String userId, String sessionId) {
     }
 
     @Override
-    public Response terminateSessionsByUserId(String userId) {
+    public Response terminateSessionsByUserId(String userId) throws UserStoreException {
+
+        try {
+            UserRealm userRealm = CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+            AbstractUserStoreManager userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+
+            String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            String adminUserName = userRealm.getRealmConfiguration().getAdminUserName();
+            String adminUserID = userStoreManager.getUserIDFromUserName(adminUserName);
+
+            if (!username.equals(adminUserName) && userId.equals(adminUserID)) {
+                return Response.status(403).build();
+            }
+
+            Util.validateUserId(SessionManagementServiceHolder.getRealmService(), userId,
+                    ContextLoader.getTenantDomainFromContext());
+            sessionManagementService.terminateSessionsByUserId(userId);
+            return Response.noContent().build();
+        } catch (NullPointerException e) {
+            return Response.status(500).build();
```

Comment:
Better to add an error log statement here. WDYT?

Reply:
Fixed in https://github.com/wso2/identity-api-user/pull/134/commits/2a89fcbf917e2a678bb4b02a619d083d8c210b84